jerry-core/parser/js/js-lexer.c

```diff
@@ -1298,26 +1298,43 @@ lexer_parse_string (parser_context_t *context_p, /**< context */
 } /* lexer_parse_string */
 
 /**
- * Parse octal number.
+ * Check number
  */
-static inline void
-lexer_parse_octal_number (parser_context_t *context_p, /** context */
-                          const uint8_t **source_p) /**< current source position */
+static void
+lexer_check_numbers (parser_context_t *context_p, /**< context */
+                     const uint8_t **source_p, /**< source_pointer */
+                     const uint8_t *source_end_p, /**< end of the source */
+                     const ecma_char_t largest_number, /**< largest valid number */
```

Comment:
comment: largest allowed number
Btw, I would prefer "digit_max" since it represents the maximum digit, rather than the result number